### PR TITLE
LAVD online

### PR DESCRIPTION
**Pull request by nathanieltarshish**
Update: computes the lavd field online and outputs it in the float files. Also adds a header to float files

----------------------------------------------------------------------------------------------------------------
Old: This is not a final version of the code. The subroutines `flt_lavd.F` and `flt_compute_global_mean.F` write out info for debugging to STDOUT. Everything looks to be working except the global mean computation. It looks like `vort3` may not be properly passed between the two subroutines or is indexed wrong in `flt_compute_global_mean.F` .  As a temporary fix, `mean_vort` is set to zero before computing LAVD.

### CODE REVIEW

**Comment by nathanieltarshish** [comment]
Here is example debug code from STDOUT
```
(PID.TID 0077.0001) ----- DEBUG COMPUTE_GLOBAL_MEAN_RL --------
(PID.TID 0077.0001) myThid=            1last temp area  =19188411.5049900077283382
(PID.TID 0077.0001) myThid=            1last temp vorticity  =       1.0000000000000000
(PID.TID 0077.0001) myThid=            1thread total vorticity*area =*************************
(PID.TID 0077.0001) myThid=            1thread total area =*************************
(PID.TID 0077.0001) ------ END DEBUG COMPUTE_GLOBAL_MEAN_RL --------
``` 
It do not think 1.0 is a realistic value for the last temporary vorticity.

**Comment by rabernat** [comment]
I would have a close look at this file:
http://wwwcvs.mitgcm.org/viewvc/MITgcm/MITgcm/pkg/monitor/mon_stats_rs.F?revision=1.9&view=markup
To see how some global means are actually computed.

**Comment by rabernat** [comment]
This is where the sum routines are defined:
http://wwwcvs.mitgcm.org/viewvc/MITgcm/MITgcm/eesupp/src/global_sum.F?revision=1.16&view=markup

**Comment by rabernat** [comment]
You might have to call `global_sum_tile` before `global_sum`
http://wwwcvs.mitgcm.org/viewvc/MITgcm/MITgcm/eesupp/src/global_sum_tile.F?revision=1.5&view=markup

File: code_lavd/flt_traj.F [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      INTEGER i, j
+      INTEGER iG, jG
+      _RL ix, jy, i0x, j0y, xx, yy, zz
+      _RL uu, vv, tt, ss, pp
+C     relative vorticity on float position
+      _RL vo
+C     relative vorticity (Eulerian field)
+      _RL vort3(1-OLx:sNx+OLx, 1-OLy:sNy+OLy, Nr)
+      _RS hFacZ(1-OLx:sNx+OLx, 1-OLy:sNy+OLy)
+      _RS r_hFacZ(1-OLx:sNx+OLx,1-OLy:sNy+OLy)
```
Comment (rabernat) [inline code_lavd/flt_traj.F]: We should update these variable sizes to all end with `Nr, nSx, nSy`

**Comment by rabernat** [comment]
Look at the rl version:
http://wwwcvs.mitgcm.org/viewvc/MITgcm/MITgcm/pkg/monitor/mon_stats_rl.F?revision=1.9&view=markup

File: code_lavd/flt_compute_global_mean.F [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+#include "CPP_OPTIONS.h"
+
+CBOP 0
+C !ROUTINE: FLT_LAVD
+
+C !INTERFACE:
+
+      SUBROUTINE COMPUTE_GLOBAL_MEAN_RL (
+     I                myNr, arr, arrMask, arrArea,
+     I                mean, bi, bj,
```
Comment (rabernat) [inline code_lavd/flt_compute_global_mean.F]: Change this to
```python
     O                mean,
     I                bi, bj
```

**Comment by nathanieltarshish** [comment]
I think it is ready now.

**Comment by rabernat** [comment]
Hard to review this line by line, since all the code is in a new folder and therefore doesn't render as a diff.

But if you say it works, I will just merge.

**Pull request merged**